### ISSUE TITLE
Update ucanlog to ucanto@7.0.0

### ISSUE THREAD

**Issue by Gozala** [web3-storage/w3infra#186]
ucanto is introducing major breaking changes in 7.0 in an attempt to avoid breaking changes in the future, specifically it's wire protocol was updated to send versioned messages between (client/server) agents, which look as follows (for more details see https://github.com/web3-storage/ucanto/pull/274)

```ts
/**
 * Describes messages send across ucanto agents.
 */
export type AgentMessageModel<T> = Variant<{
  'ucanto/message@7.0.0': AgentMessageData<T>
}>

/**
 * Describes ucanto@7.0.0 message format send between (client/server) agents.
 *
 * @template T - Phantom type capturing types of the payload for the inference.
 */
export interface AgentMessageData<T> extends Phantom<T> {
  /**
   * Set of (invocation) delegation links to be executed by the agent.
   */
  execute?: Tuple<Link<UCAN.UCAN<[Capability]>>>

  /**
   * Map of receipts keyed by the (invocation) delegation.
   */
  report?: Record<ToString<UCANLink>, Link<ReceiptModel>>
}
```

TLDR; CARs sent will contain single root pointing to the `AgentMessageModel` which is variant (a.k.a [keyed union](https://ipld.io/docs/schemas/features/representation-strategies/#union-keyed-representation)). At the moment variant has a single branch corresponding to ucanto 7.0 schema, but in the future we'll add branches as we implement different versions of invocation spec (so we can infer which version of wire protocol used).

Current version tagged as `ucanto/message@7.0.0` has links to:

1. Set of invocation delegations (which used to be the CAR roots in the requests up until ucanto 7.0).
1. Set of receipt links

While right now all requests will only contain only invocations and all responses receipts, in the future we may start dispatching invocations back to client agents and consequently also get receipts for those from the, which is why `AgentMessageData` has both fields. Both feilds are optional and are omitted if corresponding set is empty, although this is only enforced at the type level.

---

Deployed UCAN log server now receives bytes on on `/ucan/` endpoint and based on the content-type header either treats it as invocations (if `'application/invocations+car'`)  or receipts (if `'application/receipt+dag-cbor'`) https://github.com/web3-storage/w3infra/blob/abe2410fd43e7183c6d4a333277e2bd605726606/upload-api/ucan-invocation.js#L39-L57

We should update the `/ucan/` endpoint to instead to receive CARs containing `AgentMessageModel` in a root and index invocations & receipts from it instead.

**Comment by gobengo** [comment]
> We have decided that separate HTTP endpoint where invocation CID is part of the path e.g. /receipt/${invocation.cid} would be a good way to avoid leaking too much implementation details

Sounds like a good thing to have in a 'ucanto over http' spec, along with requirements around `Accept` http header usage and well-known media types therein.

**Comment by Gozala** [comment]
> what kind of bytes? CAR bytes ? @Gozala

CBOR of the `receipt.out` field
